de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/EditorAPI.java

```diff
@@ -33,26 +35,43 @@ public EditorAPI(Project project) {
   }
 
   /**
-   * Sets the given Editor to the specified line range in the UI thread.
+   * Scrolls the given editor so that the given line is in the center of the local viewport. The
+   * given line represents the logical position in the editor.
    *
-   * @param editor
-   * @param lineStart
-   * @param lineEnd
+   * <p><b>NOTE:</b> The center of the local viewport is at 1/3 for IntelliJ.
+   *
+   * @param editor the editor to scroll
+   * @param line the line to scroll to
+   * @see LogicalPosition
    */
-  public void setViewPort(final Editor editor, final int lineStart, final int lineEnd) {
+  void scrollToViewPortCenter(final Editor editor, final int line) {
+    application.invokeAndWait(
+        () -> {
+          LogicalPosition logicalPosition = new LogicalPosition(line, 0);
+
+          editor.getScrollingModel().scrollTo(logicalPosition, ScrollType.CENTER);
+        },
+        ModalityState.defaultModalityState());
+  }
 
-    Runnable action =
-        new Runnable() {
-          @Override
-          public void run() {
+  /**
+   * Returns the logical line range of the local viewport for the given editor.
+   *
+   * @param editor the editor to get the viewport line range for
+   * @return the logical line range of the local viewport for the given editor
+   * @see LogicalPosition
+   */
+  LineRange getLocalViewportRange(Editor editor) {
```

Comment:
You have the same logic as in the previous commit in file LocalViewPortChangeHandler.java. Is there an appropriate location where you could place this function as helper?

Reply:
This is the correct position as a "helper" since this class is only used to encapsulate the IntelliJ API access dealing with editor stuff. Will replace the other usage with a reference to this method.